spec/requests/api/providers_spec.rb

```diff
@@ -189,6 +219,35 @@
       expect(ExtManagementSystem.exists?(p1_id)).to be_truthy
       expect(ExtManagementSystem.exists?(p2_id)).to be_truthy
     end
+
+    it "supports provider with multiple endpoints creation" do
+      def hostname(connection)
+        connection["endpoint"]["hostname"]
+      end
+
+      def token(connection)
+        connection["authentication"]["auth_key"]
+      end
+
+      api_basic_authorize collection_action_identifier(:providers, :create)
+
+      run_post(providers_url, gen_request(:create, sample_openshift_mep))
+
+      expect(response).to have_http_status(:ok)
+      expect_result_resource_keys_to_be_like_klass("results", "id", Integer)
+      expect_results_to_match_hash("results", [sample_openshift_mep.except(*ENDPOINT_ATTRS)])
+
+      provider_id = response_hash["results"].first["id"]
+      expect(ExtManagementSystem.exists?(provider_id)).to be_truthy
+      provider = ExtManagementSystem.find(provider_id)
+
+      default_connection = sample_openshift_mep["connection_configurations"][0]
+      hawkular_connection = sample_openshift_mep["connection_configurations"][1]
+      expect(provider.connection_configurations.default.endpoint.hostname).to eq(hostname(default_connection))
```

Comment:
> i.e. if this information is available in the response body, it may be better to test against that instead, at least in a request spec.

@imtayadeway This is information (endpoints and authentications) is not included in the response for the post, it is also not included in regular get request.
We can do a get with `?attributes=authentications,endpoints` but this will not look good in the create spec ...